src/Refactoring-UI/ReExtractToTemporaryDriver.class.st

```diff
@@ -0,0 +1,106 @@
+Class {
+	#name : 'ReExtractToTemporaryDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'sourceInterval',
+		'newVariableName',
+		'selector',
+		'class',
+		'shouldEscape',
+		'variableNameEditorPresenterClass'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'execution' }
+ReExtractToTemporaryDriver >> changes [
+	
+	^ refactoring privateTransform; changes
+
+]
+
+{ #category : 'resources' }
+ReExtractToTemporaryDriver >> configureRefactoring [
+
+	refactoring := RBExtractToTemporaryRefactoring
+							extract: sourceInterval
+							from: selector
+							in: class
+]
+
+{ #category : 'instance creation' }
+ReExtractToTemporaryDriver >> extract: anInterval from: aSelector in: aClass [
+
+	class := RBNamespace new classObjectFor: aClass.
+	selector := aSelector.
+	sourceInterval := anInterval.
+]
+
+{ #category : 'instance creation' }
+ReExtractToTemporaryDriver >> extract: anInterval to: aString from: aSelector in: aClass [
+
+	class := RBNamespace new classObjectFor: aClass.
+	selector := aSelector.
+	sourceInterval := anInterval.
+	newVariableName := aString
+]
+
```

Comment:
I think we will not be using this constructor anymore, so we can remove it. This is because this will be invoked through command, and command should never give it new name.